cirq/protocols/circuit_diagram_info_protocol.py

```diff
@@ -133,6 +134,15 @@ def __repr__(self):
                                          self.use_unicode_characters,
                                          self.precision, self.qubit_map))
 
+    def format_real(self, val: Union[sympy.Basic, int, float]):
+        if isinstance(val, sympy.Basic):
+            return str(val)
+        if val == int(val):
+            return str(int(val))
+        if self.precision is None:
+            return str(val)
+        return '{{:.{}}}'.format(self.precision).format(float(val))
```

Comment:
No need to resort to calling `format`:

```python
return f'{float(val):.{self.precision}f}'
```

should work.

Reply:
Much nicer. Done.